core/provider_management.py

```
@@ -0,0 +1,674 @@
+            # Update status
+            provider.status = 'INACTIVE'
+            provider.deactivation_reason = reason
+            provider.deactivation_date = timezone.now()
+            provider.save()
+            
+            return {
+                'success': True,
+                'message': 'Provider deactivated successfully',
+                'provider_id': provider_id,
+                'reason': reason
+            }
```

Ruff (0.13.2)

Consider moving this statement to an `else` block

(TRY300)